marked.parse

Pupo1 opened the pull request:
This is my first pull request. I hope I get everything right. 

I noticed that when I try to access any board, Kanbana will crash with the following error: 
```

$ ~/kanbana/kanbana-master $ npm start

    kanbana@2.0.0 start
    node index.js

Starting server http://localhost:3000/
Cache time: 10 min
See node index.js -h for help

get(): serving main menu to ::ffff:134.19.52.89
get(): serving /board/pupo to ::ffff:134.19.52.89
/home/borg/kanbana/kanbana-master/index.js:56
body: marked(data.substring(data.indexOf(heads[i]), heads[i + 1] !== undefined ? data.indexOf(heads[i + 1]) : data.length).replace(heads[i], "")),
^

TypeError: marked is not a function
at renderHTML (/home/pupo/kanbana/kanbana-master/index.js:56:19)
at /home/borg/kanbana/kanbana-master/index.js:142:24
at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3)

Node.js v19.3.0
```

I don't have much of a clue of java script, but google tells me that the command should be marked.parse instead of marked.

SrGMC committed:
Refactored from gulp to express.js

Added small cache
Fixed a few bugs

SrGMC committed:
Update package.json

SrGMC committed:
Update Readme.md

SrGMC committed:
Update README.md

SrGMC committed:
Update README.md

SrGMC committed:
Create screenshot.png

SrGMC committed:
Update README.md

SrGMC committed:
Added TODOs

SrGMC committed:
Update README.md

SrGMC committed:
Delete ivan.md

SrGMC committed:
Fixed git deletion

SrGMC committed:
Implemented CLI arguments, 404 page and main menu

SrGMC committed:
Update README.md

SrGMC committed:
Update README.md

SrGMC committed:
Commented code

SrGMC committed:
Delete .DS_Store

SrGMC committed:
Fixed package.json

SrGMC committed:
Merge branch 'master' of https://github.com/SrGMC/crepido

SrGMC committed:
Fixed -p not working

SrGMC committed:
Fixed cache flushing not working

SrGMC committed:
Small fix at garbage collector

SrGMC committed:
Update .gitignore

SrGMC committed:
Renamed Crepido to Kanbana

SrGMC committed:
Updated .gitignore. Added example board

SrGMC committed:
Updated links

SrGMC committed:
Treat other header levels as h1

SrGMC committed:
Formatting and security updates

SrGMC committed:
Define port and cache from env vars and serve different files

SrGMC committed:
Improved responsive design

SrGMC committed:
Use OpenColor colors

SrGMC committed:
Fix header and allow empty images

SrGMC committed:
Fix empty picture

SrGMC committed:
Custom label colors

SrGMC committed:
Dark mode and remove unnecessary classes

SrGMC committed:
Fix dark mode links

SrGMC committed:
Fix title rendering

SrGMC committed:
Examples and README.md

SrGMC committed:
Docker support

SrGMC committed:
Bump up release

mend-bolt-for-github[bot] committed:
Add .whitesource configuration file

SrGMC committed:
Update README.md

SrGMC committed:
Update LICENSE

SrGMC committed:
Merge pull request #1 from SrGMC/whitesource/configure

Configure WhiteSource Bolt for GitHub

snyk-bot committed:
fix: upgrade express from 4.17.1 to 4.18.1

Snyk has created this PR to upgrade express from 4.17.1 to 4.18.1.

See this package in npm:
https://www.npmjs.com/package/express

See this project in Snyk:
https://app.snyk.io/org/srgmc/project/8425ecd1-649d-4f9e-bf06-011cc7e8b765?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade marked from 1.2.7 to 1.2.9

Snyk has created this PR to upgrade marked from 1.2.7 to 1.2.9.

See this package in npm:
https://www.npmjs.com/package/marked

See this project in Snyk:
https://app.snyk.io/org/srgmc/project/8425ecd1-649d-4f9e-bf06-011cc7e8b765?utm_source=github&utm_medium=referral&page=upgrade-pr

SrGMC committed:
Merge pull request #2 from SrGMC/snyk-upgrade-660aee6e7d9a7c7fd92e14e673dae991

[Snyk] Upgrade express from 4.17.1 to 4.18.1

SrGMC committed:
Merge branch 'master' into snyk-upgrade-2993da3b3276e5c4bc64383605cdb14e

SrGMC committed:
Fix JSON formatting

SrGMC committed:
Merge pull request #3 from SrGMC/snyk-upgrade-2993da3b3276e5c4bc64383605cdb14e

[Snyk] Upgrade marked from 1.2.7 to 1.2.9

Pupo1 commented on index.js:
```diff
@@ -0,0 +1,259 @@
+    for (var i = 0; i < heads.length; i++) {
+        boards[i] = {
+            head: JSON.parse("{" + heads[i].replace(/(---)/g, "") + "}"),
+            body: marked(data.substring(data.indexOf(heads[i]), heads[i + 1] !== undefined ? data.indexOf(heads[i + 1]) : data.length).replace(heads[i], "")),
```
body: marked.parse(data.substring(data.indexOf(heads[i]), heads[i + 1] !== undefined ? data.indexOf(heads[i + 1]) :

issmirnov commented:
Hi @Pupo1  - welcome to Github! It's so great to see your first pull request. My apologies about the delayed review, I was out on an extended winter holiday.

I'm trying to better understand this pull request - it appears that the proposal is to merge all 49 commits from https://github.com/SrGMC/kanbana into this repo - is that correct? You mentioned `body: marked.parse(data.substring(data.indexOf(heads[i]), heads[i + 1] !== undefined ? data.indexOf(heads[i + 1]) :` line, but I don't see it in the diff.

For context, this is an archival fork for some software that I used to use, but it appears that @SrGMC is maintaining a modern version at https://github.com/SrGMC/kanbana. Perhaps it's better to open an PR there?

Pupo1 commented:
Hello, 

seems I got it terribly wrong - no, I did not want to merge the other requests. My only tiny little pull request should be to write "marked.parse" instead of "marked", because otherwise there would be an error saying "marked is not a function". Seems I need to dive a little deeper into how to make pull requests... 

Best regards, 
Pupo1

On Fri, Jan 13, 2023 at 09:36:40PM -0800, Ivan Smirnov wrote:
>    Hi ***@***.*** - welcome to Github! It's so great to see your first pull
>    request. My apologies about the delayed review, I was out on an extended
>    winter holiday.
> 
>    I'm trying to better understand this pull request - it appears that the
>    proposal is to merge all 49 commits from
>    [2]https://github.com/SrGMC/kanbana into this repo - is that correct? You
>    mentioned body: marked.parse(data.substring(data.indexOf(heads[i]),
>    heads[i + 1] !== undefined ? data.indexOf(heads[i + 1]) : line, but I
>    don't see it in the diff.
> 
>    For context, this is an archival fork for some software that I used to
>    use, but it appears that ***@***.*** is maintaining a modern version at
>    [4]https://github.com/SrGMC/kanbana. Perhaps it's better to open an PR
>    there?
> 
>    —
>    Reply to this email directly, [5]view it on GitHub, or [6]unsubscribe.
>    You are receiving this because you were mentioned.Message ID:
>    ***@***.***>
> 
> References
> 
>    Visible links
>    1. https://github.com/Pupo1
>    2. https://github.com/SrGMC/kanbana
>    3. https://github.com/SrGMC
>    4. https://github.com/SrGMC/kanbana
>    5. https://github.com/issmirnov-forks/crepido/pull/1#issuecomment-1382667153
>    6. https://github.com/notifications/unsubscribe-auth/ABP4G65WOQBJC33L42ZOQRLWSI3ORANCNFSM6AAAAAATJ6KGUM

issmirnov commented:
No worries, it's a confusing workflow. Here's what you need to do:

1. Open https://github.com/SrGMC/kanbana, and click the "fork" button in the top right
2. On your fork (which will be at https://github.com/Pupo1/kanbana) edit the relevant file with the `marked.parse` fix.
3. Commit and push your code. If you do (2) through the web UI, it will automatically commit to probably `patch-1`
4. Open https://github.com/SrGMC/kanbana again, and you should see a banner at the top to open a pull request. Fill out the same details you filled out here, and @SrGMC should get back to you.

Let me know if you have any questions. And again, welcome to github and the open source world!

Pupo1 commented:
Thanks a lot for your help and info. I will try it and hope it will work next time.

snyk-bot committed:
fix: package.json & package-lock.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-MARKED-1070800
- https://snyk.io/vuln/SNYK-JS-MARKED-2342073
- https://snyk.io/vuln/SNYK-JS-MARKED-2342082

SrGMC committed:
Merge pull request #5 from SrGMC/snyk-fix-0d7ce66c158dae2bdbad91f9dda9d22a

[Snyk] Security upgrade marked from 1.2.9 to 4.0.10
